src/omv/img/filter.c

```diff
@@ -244,37 +269,54 @@ void imlib_mean_filter(image_t *img, const int ksize, bool threshold, int offset
             break;
         }
         case IMAGE_BPP_RGB565: {
+            int pixel, r, g, b, r_acc, g_acc, b_acc;
             buf.data = fb_alloc(IMAGE_RGB565_LINE_LEN_BYTES(img) * brows, FB_ALLOC_NO_HINT);
 
             for (int y = 0, yy = img->h; y < yy; y++) {
                 uint16_t *row_ptr = IMAGE_COMPUTE_RGB565_PIXEL_ROW_PTR(img, y);
                 uint16_t *buf_row_ptr = IMAGE_COMPUTE_RGB565_PIXEL_ROW_PTR(&buf, (y % brows));
 
+                r_acc = g_acc = b_acc = 0;
                 for (int x = 0, xx = img->w; x < xx; x++) {
                     if (mask && (!image_get_mask_pixel(mask, x, y))) {
                         IMAGE_PUT_RGB565_PIXEL_FAST(buf_row_ptr, x, IMAGE_GET_RGB565_PIXEL_FAST(row_ptr, x));
                         continue; // Short circuit.
                     }
-
-                    int r_acc = 0, g_acc = 0, b_acc = 0;
-
-                    for (int j = -ksize; j <= ksize; j++) {
-                        uint16_t *k_row_ptr = IMAGE_COMPUTE_RGB565_PIXEL_ROW_PTR(img,
+                    if (!mask && x > ksize && x < img->w-ksize && y >= ksize && y <= img->h-ksize) {
```

Comment:
y <= img->h-ksize ?

Shouldn't it be y < img->h-ksize ?

Reply:
Pushed a change which fixes the Y boundary checks for mean